src/sinks/gcp/chronicle_unstructured.rs

```diff
@@ -444,14 +464,88 @@ impl RequestBuilder<(String, Vec<Event>)> for ChronicleRequestBuilder {
     }
 }
 
+#[cfg(test)]
+mod tests {
+    use std::convert::TryInto;
+
+    use super::valid_label_name;
+
+    #[test]
+    fn valid_label_names() {
+        assert!(valid_label_name(&"name".try_into().unwrap()));
+        assert!(valid_label_name(&"bee-bop".try_into().unwrap()));
+        assert!(valid_label_name(&"a09b".try_into().unwrap()));
+        assert!(valid_label_name(&"09ba".try_into().unwrap()));
+        assert!(valid_label_name(&"abc--".try_into().unwrap()));
+        assert!(valid_label_name(
+            &"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+                .try_into()
+                .unwrap()
+        ));
+
+        assert!(!valid_label_name(
+            &"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+                .try_into()
+                .unwrap()
+        ));
+        assert!(!valid_label_name(&" name ".try_into().unwrap()));
+        assert!(!valid_label_name(&"bee_bop".try_into().unwrap()));
+        assert!(!valid_label_name(&"".try_into().unwrap()));
+        assert!(!valid_label_name(&" ".try_into().unwrap()));
+        assert!(!valid_label_name(&"_*".try_into().unwrap()));
+        assert!(!valid_label_name(&"-a".try_into().unwrap()));
+        assert!(!valid_label_name(&"-".try_into().unwrap()));
+        assert!(!valid_label_name(&"_".try_into().unwrap()));
+        assert!(!valid_label_name(&"*".try_into().unwrap()));
+        assert!(!valid_label_name(&"{{field}}".try_into().unwrap()));
+    }
+}
+
+// valid chars: [a-z0-9][a-z0-9-] upto 63 chars
+// See https://cloud.google.com/chronicle/docs/preview/cloud-integration/create-custom-labels#label_requirements
+fn valid_label_name(label: &String) -> bool {
```

Comment:
This function could be simplified by matching against a regex. Especially if it's run at start up time we don't need to worry about the performance.